crates/fluvio-protocol-derive/src/ast/prop.rs

```diff
@@ -151,76 +143,322 @@ impl UnnamedProp {
         }
     }
 }
-
-pub fn validate_versions(min: i16, max: Option<i16>, field: Option<&str>) -> Option<String> {
-    match (max, field) {
-        // Print name in named fields
-        (Some(max), Some(field)) if min > max => Some(format!(
-            "On {field}, max version({max}) is less than min({min})."
-        )),
-        // No name to print in unnamed fields
-        (Some(max), None) if min > max => {
-            Some(format!("Max version({max}) is less than min({min})."))
-        }
-        (None, Some(field)) if min < 0 => {
-            Some(format!("On {field} min version({min}) must be positive."))
+/// Convert the values to TokenStream which will be ready to use variable value
+///
+/// # Example
+/// ````ignore
+/// // Function as a literal
+/// fn test() -> i16 { 1 }
+/// #[fluvio(min_version = "test()")]
+/// ````
+/// To use the value from the test() function:
+/// ````ignore
+/// let func_value = prop_attrs_type_value(prop_attr_type, None)
+/// ````
+/// To set a specific type you can do this:
+/// ````ignore
+/// let ident_type = Ident::new("u8", Span::call_site());
+/// let func_value = prop_attrs_type_value(prop_attr_type, Some(&ident_type))
+/// ````
+///
+pub fn prop_attrs_type_value(
+    attrs_type: Option<&PropAttrsType>,
+    ident_type: Option<&Ident>,
+) -> TokenStream {
+    if let Some(attr) = attrs_type {
+        match &attr {
+            PropAttrsType::Lit(data) => parse_quote!(#data),
+            PropAttrsType::Fn(data) => TokenStream::from_str(&format!("{}()", data)).unwrap(),
+            PropAttrsType::Int(data) => {
+                if let Some(itype) = ident_type {
+                    TokenStream::from_str(&format!("{}_{}", data, itype)).unwrap()
```

Comment:
Need to find a way without a need for `unwrap` or properly propagate the error

Reply:
I am not really sure if this is an issue to be unwrap here as this is just a TokenStream and if anything happens it will be in compile time. Maybe we can just use `expect` insteead of `unwrap`?

Reply:
https://docs.rs/syn/latest/syn/struct.Error.html

Reply:
I don't think we can convert LexError with compile_error. From what I understand LexError already emits any errors but if we just want to remove the `unwrap` maybe we can do it like this.

`TokenStream::from_str(&format!("{}_i16", data)).map_err(|_| TokenStream::new())`

This way we don't have unwrap and still will be having all the errors.